server/controllers/file.js

```diff
@@ -0,0 +1,258 @@
+            if (file) {
+                const fileName = file.files[0].file_name;
+                const encryptedPath = '../server/private/' + fileName;
+                const decryptedPath = '../server/public/' + fileName;
+                await getFile(cid, encryptedPath).then(async() => {
+                    try {
+                        jscrypt.decryptFile(
+                            encryptedPath,
+                            decryptedPath,
+                            "aes256",
+                            file.encryption_key,
+                            655000,
+                            (isDone) => {
+                                if (isDone === true) {
+                                    console.log(fileName + ' is decrypted successfully!');
+                                    console.log("Sending files to the user");
+                                    //send the file to the client
+                                    res.sendFile(path.resolve(decryptedPath));
+    
+                                    setTimeout(() => {
+                                        unlink(decryptedPath, (err) => {
+                                            if (err) {
+                                                console.log(err);
+                                            }
+                                            console.log(decryptedPath + ' is deleted!');
+                                        })
+    
+                                        unlink(encryptedPath, (err) => {
+                                            if (err) {
+                                                console.log(err);
+                                            }
+                                            console.log(encryptedPath + ' is deleted!');
+                                        })
+                                    }, 2 * 60 * 1000)
+    
+                                }
+                                else {
+                                    console.log("File decryption in progress...");
+                                }
+                            }
+                        )
+                    } catch (err) {
+                        console.log(err);
+                        return res.status(500).json({ error: err.message });
+                    }
+                })
+            } else {
+                return res.status(200).sendFile("../private/hacker.png", { root: __dirname });
+            }
+        } catch (err) {
+            return res.status(500).json({ error: err.message });
+        }
+    }
+
+    static async download(req, res){
+        const cid = req.params.cid;
+        try {
+            const file = await User.findOne({ files: { $elemMatch: { cid: cid, public: true } } }, { encryption_key: 1 }).select({ files: { $elemMatch: { cid: cid, public: true } } });
```

Comment:
Same for here too.
![image](https://user-images.githubusercontent.com/63467479/192140959-8863841b-81cd-4003-aa23-0d78263e09e2.png)